ambry-store/src/main/java/com/github/ambry/store/DiskManager.java

```diff
@@ -494,7 +494,9 @@ boolean removeBlobStore(PartitionId id) {
   private Map<String, Throttler> getThrottlers(StoreConfig config, Time time) {
     Map<String, Throttler> throttlers = new HashMap<>();
     // compaction ops
-    Throttler compactionOpsThrottler = new Throttler(config.storeCompactionOperationsBytesPerSec, -1, true, time);
+    Throttler compactionOpsThrottler =
+        new Throttler(config.storeCompactionOperationsBytesPerSec, config.storeCompactionThrottlerCheckIntervalMs, true,
```

Comment:
could you elaborate why adding a positive checkIntervalMs would change behavior of Throttler?

Reply:
with -1, the rated is check every call. 
We also have a similar metric, `compactionCopyRateInBytes`, which shows stats per 60 seconds. 
Currently, if we set `storeCompactionOperationsBytesPerSec` to 5MB, the `compactionCopyRateInBytes` may go up to 15MB/s.
I suspect the difference come from the `-1`.